### ISSUE TITLE
`tsconfig` instead of `jsconfig` breaks VSCode IntelliSense imports on JavaScript project

### ISSUE THREAD

**Issue by dalisoft** [microsoft/vscode#139964]
Issue Type: <b>Bug</b>

I recently noticed IntelliSense imports stopped working and investaged why, appears i just changed `jsconfig.json` to `tsconfig.json`. Reverting makes imports work back. On WebStorm even with `tsconfig.json` works fine

VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:38:17.605Z)
OS version: Darwin arm64 21.2.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.08GB free)|
|Process Argv|--crash-reporter-id 0f6b11bf-04f7-4592-a1e4-016b55459da1|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (62)</summary>

Extension|Author (truncated)|Version
---|---|---
svgo|100|2.6.0
vscode-fish|bma|1.0.24
vscode-intelephense-client|bme|1.8.0
npm-intellisense|chr|1.4.0
vscode-css-modules|cli|0.4.1
gitignore|cod|0.7.0
vscode-svgviewer|css|2.0.0
vscode-markdownlint|Dav|0.45.0
vscode-eslint|dba|2.2.2
gitlens|eam|11.7.0
EditorConfig|Edi|0.16.4
vscode-npm-script|eg2|0.3.23
LogFileHighlighter|emi|2.13.0
prettier-vscode|esb|9.0.0
shell-format|fox|7.2.2
vscode-google-translate|fun|1.4.13
remotehub|Git|0.20.0
vscode-pull-request-github|Git|0.34.2
go|gol|0.30.0
todo-tree|Gru|0.0.214
vscode-phpfmt|kok|1.0.30
sftp|lix|1.12.10
rust-analyzer|mat|0.2.867
vscode-language-babel|mgm|0.0.35
git-graph|mhu|1.30.0
dotenv|mik|1.0.1
HTMLHint|mka|0.10.0
vscode-apache|mrm|1.2.0
vscode-scss|mrm|0.10.0
gather|ms-|2021.9.1
python|ms-|2021.12.1559732655
vscode-pylance|ms-|2021.12.2
jupyter|ms-|2021.11.1001550889
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.4
remote-containers|ms-|0.209.6
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
remote-wsl|ms-|0.63.13
vscode-remote-extensionpack|ms-|0.21.0
cpptools|ms-|1.7.1
color-highlight|nau|2.5.0
proto|pet|0.0.2
material-icon-theme|PKi|4.11.0
polacode|pnp|0.3.4
vscode-thunder-client|ran|1.9.1
java|red|1.2.0
vscode-commons|red|0.0.6
vscode-yaml|red|1.2.2
postcss|ric|2.0.0
code-settings-sync|Sha|3.4.3
gitconfig|sid|2.0.1
code-spell-checker|str|2.0.14
code-spell-checker-russian|str|2.0.2
vscode-styled-components|sty|1.7.2
vscode-stylelint|sty|1.2.1
vscode-sql-lit|the|0.4.0
vscodeintellicode|Vis|1.2.15
vscode-nginx|wil|0.7.2
viml|Xad|1.0.1
local-history|xyz|1.8.1
vscode-open-in-github|ziy|1.3.6

</details>

<details>

<summary>jsconfig.json</summary>

```json
{
  "compilerOptions": {
    "jsx": "react",
    "baseUrl": "src"
  },
  "include": [
    "src"
  ]
}
```
</details>

**Comment by IllusionMH** [comment]
/needsMoreInfo

Did you just rename `jsconfig.json` to `tsconfig.json` without changes?

Only imports for `.js` files are broken or `.ts` files as well?

`jsconfig.json` implies `"allowJs": true` and therefore you need to set this manually in `tsconfig.json`.

**Comment by dalisoft** [comment]
- Yes, changed without changes
- Yes, both file types are broken
- I enabled allowJs too as recently wanted to migrate to Typescript

**Comment by IllusionMH** [comment]
1 and 3 are contradicting either there are no changes (and `allowJS` default to `false`) or it was updated somehow.

You should check that all files (`.js` and `.ts`) are handled by same config to have them suggested by auto import. You can do that by opening files and checking language status (see https://code.visualstudio.com/updates/v1_61#_new-javascript-and-typescript-language-status-item)

Also is it reproducible with all extensions disabled?
You can try this with <kbd>F1</kbd> and `>Developer: Reload Window With Extensions Disabled`

**Comment by dalisoft** [comment]
@IllusionMH Thank you for helping.
- I did not changed files from `.js` to `.ts` yet, just want test 
- Then found importing functions/React components IntelliSense does not work properly (taked two days to found 😄 )
- For example of `compilerOptions`, take a look for [this](https://github.com/dalisoft/airlight/blob/master/tsconfig.json)

**Comment by dalisoft** [comment]
My bad, seems `tsconfig` does not share `include` and we should not add `node_modules` as exclude for properly working if there any dependent shared files, we can close this issue as wrong. Thank you for helping

TLDR:
- Always write (duplicate) `include` like (`include: ["src", "node_modules/YOUR_SHARED_THING"]`)
- Don't write `exclude` on `tsconfig.json`
- Enable `allowJs` always (i did this already)

Happy migrating and happy new year!
